lta/bundler.py

```diff
@@ -265,10 +282,25 @@ def __init__(self, config: Dict[str, str], logger: Logger) -> None:
     return bundler.lta_ok
 
 
+async def lifecycle_loop(bundler: Bundler) -> None:
+    """Run a check for a stop semaphore as an infinite loop."""
+    bundler.logger.info("Starting lifecycle loop")
+    while True:
+        # if there is a stop semaphore, terminate the program
+        if check_stop_semaphore():
+            bundler.logger.info("Component stopped; shutting down.")
+            sys.exit(0)
+        # sleep until we check again
+        await asyncio.sleep(1.0)
+
+
 async def status_loop(bundler: Bundler) -> None:
     """Run status heartbeat updates as an infinite loop."""
     bundler.logger.info("Starting status loop")
     while True:
+        # if there is a drain semaphore, stop sending status updates
+        if check_drain_semaphore():
+            break
```

Comment:
Could you just do:

```python
while not check_drain_semaphore():
```

Reply:
I liked this better, so I did end up using it verbatim.